src/murmur/MurmurGRPCImpl.cpp

```diff
@@ -138,10 +140,18 @@ MurmurRPCImpl::MurmurRPCImpl(const QString &address, std::shared_ptr<::grpc::Ser
 	m_completionQueue = builder.AddCompletionQueue();
 	m_server = builder.BuildAndStart();
 	meta->connectListener(this);
+	b_IsRunning = true;
 	start();
 }
 
 MurmurRPCImpl::~MurmurRPCImpl() {
+	void *ignored_tag;
+	bool ignored_ok;
+	b_IsRunning = false;
+	m_server->Shutdown(std::chrono::system_clock::now());
+	m_completionQueue->Shutdown();
+	while (m_completionQueue->Next(&ignored_tag, &ignored_ok))
```

Comment:
Should delete the drained tags instead of leaking them. No reason to needlessly create valgrind/asan false positives even if it doesn't endanger application stability.

Reply:
I can delete these. I didn't because the server is about to be shut down and everything is going to be freed anyway.